cmd/fastly/main.go

```diff
@@ -78,6 +79,17 @@ func main() {
 		}
 	}
 
+	if file.CLI.Version != revision.SemVer(revision.AppVersion) {
+		err := file.Load(config.RemoteEndpoint, httpClient)
+		if err != nil {
+			errors.RemediationError{
+				Inner:       err,
+				Remediation: errors.NetworkRemediation,
+			}.Print(os.Stderr)
+			os.Exit(1)
+		}
+	}
```

Comment:
This is almost identical to the code above. 
Perhaps move the code above out and add `err != nil` to this condition?

Reply:
I wasn't sure I understood what you were suggesting, so I'm going to just clarify the two distinct flows that are defined, and which I don't believe can be combined (if that's what you were asking)...

**The first scenario** we're addressing begins on L55, where it attempts to read configuration from the file system and if there is no existing local config, then it will fetch the config from the remote endpoint.

**The second scenario is**: if there is an existing config, then the error block on L56 won't be executed. So on L82 we check whether the CLI has been updated recently, and if it has, then there is the _possibility_ that the existing config data we have is outdated (e.g. maybe the data is missing a new field which the updated CLI code is going to soon run and expect to exist). So in that case we'll refetch the config from the remote endpoint. 

> **NOTE**: The second load of remote configuration (scenario 2) won't be executed if the first load of remote configuration (scenario 1) did happen because when we fetch the remote configuration we update the `file.CLI.Version` and so the conditional on L82 won't pass.

Reply:
My suggestion was to write the code combining the two file.Load() together:

```
	if err != nil {
		if verboseOutput {
			if err == config.ErrLegacyConfig {
				text.Output(out, `
					Found your local configuration file (required to use the CLI) was using a legacy format.
					File is being upgraded now.
				`)
			} else {
				text.Output(out, `
					Unable to locate a local configuration file (required to use the CLI).
					File is being created now.
				`)
			}
			text.Break(out)
		}
	}

	if err != nil || file.CLI.Version != revision.SemVer(revision.AppVersion) {
		err := file.Load(config.RemoteEndpoint, httpClient)
		if err != nil {
			errors.RemediationError{
				Inner:       err,
				Remediation: errors.NetworkRemediation,
			}.Print(os.Stderr)
			os.Exit(1)
		}
	}
```

Reply:
Ah I see, OK cool 👍🏻